pinot-common/src/main/java/org/apache/pinot/common/utils/LLCSegmentName.java

```diff
@@ -59,6 +59,28 @@ public LLCSegmentName(String tableName, int partitionGroupId, int sequenceNumber
     _segmentName = tableName + SEPARATOR + partitionGroupId + SEPARATOR + sequenceNumber + SEPARATOR + _creationTime;
   }
 
+  private LLCSegmentName(String tableName, int partitionGroupId, int sequenceNumber, String creationTime,
+      String segmentName) {
+    _tableName = tableName;
+    _partitionGroupId = partitionGroupId;
+    _sequenceNumber = sequenceNumber;
+    _creationTime = creationTime;
+    _segmentName = segmentName;
+  }
+
+  /**
+   * Returns the {@link LLCSegmentName} for the given segment name, or {@code null} if the given segment name does not
+   * represent an LLC segment.
+   */
+  @Nullable
+  public static LLCSegmentName getLLCSegmentName(String segmentName) {
+    String[] parts = StringUtils.splitByWholeSeparator(segmentName, SEPARATOR);
+    if (parts.length != 4) {
+      return null;
+    }
+    return new LLCSegmentName(parts[0], Integer.parseInt(parts[1]), Integer.parseInt(parts[2]), parts[3], segmentName);
```

Comment:
This method is added to avoid splitting the same string multiple times and creating extra garbage (it can avoid the extra check of segment name format before creating the `LLCSegmentName` object), so we need a new constructor without any extra processing.
Changed the static method to `of()`

Reply:
That's a valid point. Let's remove the duplicate code then. I suggest to modify private constructor to:
```java
private LLCSegmentName(String tableName, int partitionGroupId, int sequenceNumber, String creationTime, bool validate) {
    String[] parts = StringUtils.splitByWholeSeparator(segmentName, SEPARATOR);
    if (validate) {
        Preconditions.checkArgument(parts.length == 4, "Invalid LLC segment name: %s", segmentName);
    }
    _tableName = parts[0];
    _partitionGroupId = Integer.parseInt(parts[1]);
    _sequenceNumber = Integer.parseInt(parts[2]);
    _creationTime = parts[3];
    _segmentName = segmentName;
}
```
Then in the main constructor calls this with true and in the static method it's called it with false.

Reply:
This won't work because we still need to split the segment name twice in the static method. Changed the private one to directly take the split parts so that it can be reused.

Reply:
Sorry I made mistake in the method arguments. I meant:
```java
  public LLCSegmentName(String segmentName) {
    this(segmentName, true);
  }

  public static LLCSegmentName of(String segmentName) {
    return new LLCSegmentName(segmentName, false);
  }
  
  private LLCSegmentName(String segmentName, boolean validate) {
    String[] parts = StringUtils.splitByWholeSeparator(segmentName, SEPARATOR);
    if (validate) {
      Preconditions.checkArgument(parts.length == 4, "Invalid LLC segment name: %s", segmentName);
    }
    _tableName = parts[0];
    _partitionGroupId = Integer.parseInt(parts[1]);
    _sequenceNumber = Integer.parseInt(parts[2]);
    _creationTime = parts[3];
    _segmentName = segmentName;
  }
```
This doesn't do split twice, does it?

Reply:
Then `of()` will also throw exception instead of returning `null`. Within the `of()` we need to split first, then decide whether to return `null` or proceed

Reply:
My bad 😅 You're right.